### ISSUE TITLE
Vice-county information is not checked for geographical consistency

### ISSUE THREAD

**Issue by sacrevert** [NERC-CEH/irecord-app#133]
I was able to submit a record in Berkshire associated with the wrong vice-county

**Comment by kazlauskis** [comment]
Yes, we don't validate the vice-county on the client yet. This is due to the complexity of storing VCs boundaries in the app. But it is nice that you have logged this requirement here.

**Comment by japonicus** [comment]
Although precise vc-gridref validation might be problematic to implement off-line in a mobile app, it's something that could be checked easily if an internet connection was available.  There are api's that provide the county, given a grid-reference.

e.g.
https://database.bsbi.org/xmlgridrefparser.php?ref=SD45
(it would be easy to provide an equivalent JSON end-point if that would be useful, though iRecord/Indicia may already provide this as a service)

At a coarse scale it might be possible to implement vc validation in-app even while offline, using a lookup-table with county values for each hectad. I'm not sure if that would be useful.

(I've still not investigated the feasibility of a full-resolution offline vc-lookup library, but plan to one day)

**Comment by sacrevert** [comment]
Not sure if it would be useful, but I have a 1.3 million row table with the 6 fig grid refs that straddle grid boundaries, with the acceptable VCs in two separate columns. This was originally from Mark Hill for checking grid ref:VC attributions in Access. (Of course we also have the monad lookups as tables too (with areas)).
